src/index.ts

```diff
@@ -21,19 +19,29 @@ const isSSR =
 
 const useIsomorphicLayoutEffect = isSSR ? useEffect : useLayoutEffect
 
-export interface UseStore<T extends State> {
+export type UseStore<
+  T extends State,
+  CustomStoreApi extends StoreApi<T> = StoreApi<T>
+> = {
   (): T
   <U>(selector: StateSelector<T, U>, equalityFn?: EqualityChecker<U>): U
-  setState: SetState<T>
-  getState: GetState<T>
-  subscribe: Subscribe<T>
-  destroy: Destroy
-}
+  setState: CustomStoreApi['setState']
+  getState: CustomStoreApi['getState']
+  subscribe: CustomStoreApi['subscribe']
+  destroy: CustomStoreApi['destroy']
```

Comment:
If I'm not wrong, these four fields are already in `CustomStoreApi`, right?
Why not just remove these four lines?